electron/src/window.ts

```diff
@@ -48,10 +51,27 @@ export function createMainWindow() {
 
   window.loadURL(webappURL)
 
+  window.webContents.session.setPermissionRequestHandler((webContents, permission, callback) => {
+    if (!webContents.getURL().startsWith("file://") && (permission === "media" || permission === "openExternal")) {
+      return callback(false)
+    } else {
+      return callback(true)
+    }
+  })
+
   // subscribes to window.open and <a target="_blank"></a> links and opens the url in the browser
   window.webContents.on("new-window", (event, url) => {
     event.preventDefault()
-    shell.openExternal(url)
+    if (window.webContents.getURL().startsWith("file://")) {
+      shell.openExternal(url)
+    }
+  })
+
+  window.webContents.on("will-navigate", event => {
+    // limit navigation flows to unstrusted origins
+    if (!window.webContents.getURL().startsWith("file://")) {
```

Comment:
@andywer do you think this makes sense in our case? I'm not sure if I missed the point, the [example](https://github.com/doyensec/electronegativity/wiki/LIMIT_NAVIGATION_JS_CHECK) check is using a `https://...` url. 
So basically what I did should restrict the navigation to only `file://...` urls which should make sense because to open third-party URLs we always use `shell.openExternal`.

Reply:
Makes sense, but haven't tested it yet.